src/components/timers/Countdown.tsx

```diff
@@ -0,0 +1,189 @@
+import Button from '../generic/Button';
+import DisplayTime from '../generic/DisplayTime';
+import Inputs from '../generic/Inputs';
+import Panel from '../generic/Panel';
+
+import { useEffect, useRef, useState } from 'react';
+import { getDisplayHundredths, getDisplayMinutes, getDisplaySeconds } from '../../utils/helpers';
+
+interface CountdownProps {
+    onChange?: (config: {
+        workTime: { minutes: number; seconds: number };
+        isValid: boolean;
+    }) => void;
+    newTimer?: boolean; // Determines if this is a new timer being configured
+    workoutTimer?: boolean; // Determines if this is a timer being controlled by the workout
+    workTime?: { minutes: number; seconds: number }; // Work time configuration
+    elapsedTime?: number; // Elapsed time in milliseconds provided in workout context
+    active?: boolean; // Determines if the currently active timer in a workout
+    state?: 'not running' | 'running' | 'paused' | 'completed';
+}
+
+const Countdown: React.FC<CountdownProps> = ({ workTime = { minutes: 0, seconds: 0 }, elapsedTime = 0, onChange, newTimer = false, workoutTimer = false, active = false, state = 'not running' }) => {
+    const [inputMinutes, setInputMinutes] = useState(workTime.minutes);
+    const [inputSeconds, setInputSeconds] = useState(workTime.seconds);
+    const [totalMilliseconds, setTotalMilliseconds] = useState(workoutTimer ? workTime.minutes * 60000 + workTime.seconds * 1000 : 0);
+    const [isRunning, setIsRunning] = useState(false);
```

Comment:
Same here - think about if you can refactor `isRunning`, `isPaused`, and `isCompleted` into a single state hook